server/lib/src/utils.rs

```diff
@@ -74,6 +75,20 @@ pub fn readable_password_from_random() -> String {
     )
 }
 
+pub fn str_join(set: &BTreeSet<String>) -> String {
+    let alloc_len = set.iter().fold(0, |acc, s| acc + s.len() + 1);
+    let mut buf = String::with_capacity(alloc_len);
+    set.iter().for_each(|s| {
+        buf.push_str(s);
+        buf.push(' ');
+    });
+
+    // Remove the excess trailing space.
+    let _ = buf.pop();
+
+    buf
```

Comment:
(this is likely wrong due to ownership but you get the idea)

```suggestion
   set.iter().collect::<Vec<String>>().join(",")
```

join doesn't seem to be implemented on BTreeSet

Reply:
Yeah, but this is btreeset to allow fast merging of scopes in oauth2. So I think we leave it as is :(